### PR TITLE
Bump k8s deps to v1.24.3

### LINKED ISSUE #404: Update K8S Deps to 1.24
Update the k8s dependencies to 1.24, and release 1.23 version;

### PR DESCRIPTION
**Pull request by yibozhuang**
#### What this PR does / why we need it:

- Bump k8s deps to v1.24.3 on master
- Adapt to the new interface of PreFilter
- Fixed a followup in #405

#### Which issue(s) this PR fixes:

Fixes #404 

#### Special notes for your reviewer:

```release-note
Bump k8s deps to v1.24.3 on master.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/scheduler-plugins
 
-go 1.17
+go 1.18
```
Comment (Huang-Wei) [inline go.mod:3]: Any particular reason for bumping go version?

Comment (yibozhuang) [reply go.mod]: Sorry I forgot 1.24 is not on 1.18 yet, fixed.

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks @yibozhuang !

(I updated the description to include more details)

**Pull request merged**